packages/testlab/src/validate-api-spec.ts

```diff
@@ -28,5 +23,9 @@ export async function validateApiSpec(spec: OpenApiSpec): Promise<void> {
     throw new Error('Missing required property: paths at #/');
   }
 
-  await SwaggerParser.validate(spec, opts);
+  try {
+    await promisifiedValidator(spec, opts);
+  } catch (err) {
+    throw new Error(err);
```

Comment:
I find this suspicious. Are you wrapping `err` in a new `Error` object because it's a plain string? If that's the case, then you should add a check to avoid wrapping `Error` instances (e.g. `TypeError`) and losing the original stack trace.

```ts
} catch (err) {
  if (err instanceof Error) {
    throw err;
  } else {
    throw new Error(err);
  }
}
```
 
Eventually, we should fix the upstream module `swagger2openapi` to correctly throw Error instances instead of string error messages.

Reply:
Hopefully all instances of this should have been fixed in `swagger2openapi` some time ago, but just a heads-up that in the upcoming `v4.0.0` release, `swagger2openapi` will declare, export and throw a `S2OError` class so you will be able to identify intentionally thrown errors from any other problems.

Reply:
Thank you @MikeRalphson for chiming in. We have moved away from `swagger2openapi` and use `oas-validator` instead, see https://github.com/Mermade/oas-kit/tree/master/packages/oas-validator and https://www.npmjs.com/package/oas-validator

Unfortunately, it seems that we are still rewrapping errors in a new Error object :(

https://github.com/strongloop/loopback-next/blob/d2b79f6224db47b3d3a738eb3aa4c4276afb49b9/packages/testlab/src/validate-api-spec.ts#L15-L19